keyboards/kprepublic/bm40hsrgb/keymaps/coffee/keymap.c

```diff
@@ -4,16 +4,16 @@
 #include QMK_KEYBOARD_H
 
 enum custom_keycodes {
-    NULLKEY = SAFE_RANGE,    // An empty key to start (and maybe end) the enum
+    NULLKEY = SAFE_RANGE, // An empty key to start (and maybe end) the enum
     #ifdef DYNAMIC_MACRO_ENABLE
-        MCR_PLY,             // Macro play
-        MCR_REC,             // Macro record
-        MCR_SWT,             // Swap active macro
-    #endif
+    MCR_PLY,              // Macro play
+    MCR_REC,              // Macro record
+    MCR_SWT,              // Swap active macro
+    #endif // DYNAMIC_MACRO_ENABLE
     #ifdef MOUSEKEY_ENABLE
-        MS_ACL_U,
-        MS_ACL_D,
-    #endif
+    MS_ACL_U,             // Mouse speed up
+    MS_ACL_D,             // Mouse speed down
+    #endif // MOUSEKEY_ENABLE
```

Comment:
don't need a null key here. Also, generally, we recommend not hiding keycodes behind a define.  it is fine to have unused keys. Also, would prefer that QK_USER is used, rather than SAFE_RANGE, as it's more explicit.

```suggestion
    MCR_PLY = QK_USER,    // Macro play
    MCR_REC,              // Macro record
    MCR_SWT,              // Swap active macro
    MS_ACL_U,             // Mouse speed up
    MS_ACL_D,             // Mouse speed down
```
Alternatively, you could do something cursed like this:
```suggestion
    NULLKEY = (QK_USER - 1), // An empty key to start (and maybe end) the enum
    #ifdef DYNAMIC_MACRO_ENABLE
    MCR_PLY,              // Macro play
    MCR_REC,              // Macro record
    MCR_SWT,              // Swap active macro
    #endif // DYNAMIC_MACRO_ENABLE
    #ifdef MOUSEKEY_ENABLE
    MS_ACL_U,             // Mouse speed up
    MS_ACL_D,             // Mouse speed down
    #endif // MOUSEKEY_ENABLE
```

This would cause the next entry in the num to properly start at QK_USER. 

And for reference, `QK_USER` is unused, so you can use it, without issue

Reply:
My main reason for using `ifdefs` was so that when the feature wasn't in use I could have it be set as KC_NO (Which is how my rgb_matrix functions know when to turn off a key's light). I just tested your first option and it works but github won't let me commit it so I'll just push another commit with your changes if that's alright